lib/tests/tests.c

```diff
@@ -1649,6 +1591,473 @@ test_dtwf_multi_locus_simulation(void)
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_edge_table_add_row(&tables.edges, 0, 0.5, 5, 3);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_node_table_add_row(&tables.nodes, 0, 0.25, 0, TSK_NULL, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    
+    ret = tsk_edge_table_add_row(&tables.edges, 0, 1, 6, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_edge_table_add_row(&tables.edges, 0.5, 1, 6, 4);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_node_table_add_row(&tables.nodes, 0, 0.5, 0, TSK_NULL, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    
+    ret = tsk_edge_table_add_row(&tables.edges, 0, 1, 7, 5);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_edge_table_add_row(&tables.edges, 0, 1, 7, 6);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_node_table_add_row(&tables.nodes, 0, 1, 0, TSK_NULL, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    
+    ret = tsk_mutation_table_add_row(&tables.mutations, 0, 0, -1, "C", 1, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_mutation_table_add_row(&tables.mutations, 1, 1, -1, "C", 1, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_mutation_table_add_row(&tables.mutations, 2, 3, -1, "C", 1, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_mutation_table_add_row(&tables.mutations, 3, 0, -1, "C", 1, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_mutation_table_add_row(&tables.mutations, 4, 5, -1, "C", 1, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    
+    ret = tsk_site_table_add_row(&tables.sites, 0.1, "A", 1, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_site_table_add_row(&tables.sites, 0.2, "A", 1, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_site_table_add_row(&tables.sites, 0.3, "A", 1, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_site_table_add_row(&tables.sites, 0.4, "A", 1, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_site_table_add_row(&tables.sites, 0.45, "A", 1, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    
+    tsk_population_table_add_row(&tables.populations, NULL, 0);
+    
+    ret = tsk_treeseq_init(&ts, &tables, TSK_BUILD_INDEXES);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    
+    for (i = 0; i < 0; i++) {
+        ll_exact = log(rho[i]) - (3 + 3 * rho[i]) * 0.1;
+        ll_exact -= (6 + 3 * rho[i]) * 0.15;
+        ll_exact -= (3 + 2.5 * rho[i]) * 0.25;
+        ll_exact -= (1 + 2 * rho[i]) * 0.5;
+        ret = msp_log_likelihood_arg(&ts, rho[i], &lik);
+        CU_ASSERT_EQUAL_FATAL(ret, 0);
+        CU_ASSERT_DOUBLE_EQUAL(ll_exact, lik, tol);
+    }
+    for (i = 0; i < 0; i++) {
+        ll_exact = (5 * log(tree_length * theta[i]) - tree_length * theta[i]);
+        ll_exact -= 2 * log(4 * tree_length);
+        ll_exact -= 2 * log(tree_length);
+        ll_exact += log(3 / (4 * tree_length));
+        ret = msp_unnormalised_log_likelihood_mut(&ts, theta[i], &lik);
+        CU_ASSERT_EQUAL_FATAL(ret, 0);
+        CU_ASSERT_DOUBLE_EQUAL(ll_exact, lik, tol);
+    }
+    // Second test case: two MRCAs
```

Comment:
Maybe split these into three separate functions? Would be easier to follow then I think.

Reply:
Do you mean a separate test function for each test case, in which case there would be five? Or some other split?

Reply:
If the tests are fairly independent, then it'd be good to split them up into separate functions. Five seems good --- good to know we're covering different situations.